### ISSUE TITLE
org.elasticsearch.common.settings.Settings.processSetting causes "GET /_cluster/settings/" throwing ClassCastException

### ISSUE THREAD

**Issue by kkewwei** [elastic/elasticsearch#43791]
ES_VERSION: 5.6.8 (all likely in all version)
JVM version : JDK1.8.0_112
OS version : linux
Description of the problem including expected versus actual behavior:
Firstly I push several cluster settings to cluster twice:
step 1:
```
PUT /_cluster/settings
{
    "transient": {
        "logger.org.elasticsearch" :"info",
        "logger.org.elasticsearch.transport.netty":"ERROR"
    }
}
```
step 2:
```
PUT /_cluster/settings
{
    "transient": {
        "logger.org.transport.netty":"ERROR"
    }
}
```
it is executed successfully(those may be don't work, which we ignore)
Secondly I want to see the settings by "GET /_cluster/settings", What is surprising is that, it throws Exception:
```
      "root_cause": [
         {
            "type": "class_cast_exception",
            "reason": "java.lang.String cannot be cast to java.util.Map"
         }
      ]
```

Related logs on es server:
```
[WARN ][r.suppressed             ] path: /_cluster/settings, params: {pretty=}
java.lang.ClassCastException: java.lang.String cannot be cast to java.util.Map
        at org.elasticsearch.common.settings.Settings.processSetting(Settings.java:146) ~[elasticsearch-5.6.8.jar:5.6.8]
        at org.elasticsearch.common.settings.Settings.processSetting(Settings.java:166) ~[elasticsearch-5.6.8.jar:5.6.8]
        at org.elasticsearch.common.settings.Settings.processSetting(Settings.java:166) ~[elasticsearch-5.6.8.jar:5.6.8]
        at org.elasticsearch.common.settings.Settings.processSetting(Settings.java:166) ~[elasticsearch-5.6.8.jar:5.6.8]
        at org.elasticsearch.common.settings.Settings.getAsStructuredMap(Settings.java:131) ~[elasticsearch-5.6.8.jar:5.6.8]
        at org.elasticsearch.common.settings.Settings.toXContent(Settings.java:625) ~[elasticsearch-5.6.8.jar:5.6.8]
        at org.elasticsearch.rest.action.admin.cluster.RestClusterGetSettingsAction.renderResponse(RestClusterGetSettingsAction.java:90) ~[elasticsearch-5.6.8.jar:5.6.8]
        at org.elasticsearch.rest.action.admin.cluster.RestClusterGetSettingsAction.access$000(RestClusterGetSettingsAction.java:43) ~[elasticsearch-5.6.8.jar:5.6.8]
        at org.elasticsearch.rest.action.admin.cluster.RestClusterGetSettingsAction$1.buildResponse(RestClusterGetSettingsAction.java:66) ~[elasticsearch-5.6.8.jar:5.6.8]
        at org.elasticsearch.rest.action.admin.cluster.RestClusterGetSettingsAction$1.buildResponse(RestClusterGetSettingsAction.java:63) ~[elasticsearch-5.6.8.jar:5.6.8]
        at org.elasticsearch.rest.action.RestBuilderListener.buildResponse(RestBuilderListener.java:38) ~[elasticsearch-5.6.8.jar:5.6.8]
        at org.elasticsearch.rest.action.RestResponseListener.processResponse(RestResponseListener.java:37) ~[elasticsearch-5.6.8.jar:5.6.8]
```
Accord to those logs, I knew that there may be some thing wrong in Function `processSetting`, then i tested it such as:
```
processSetting(map, "", "logger.org.elasticsearch", "info");
processSetting(map, "", "logger.org.elasticsearch.transport.netty4", "ERROR");
processSetting(map, "", "logger.org.transport.netty4", "ERROR");
```
and it threw the exception as I mentioned above.
Reson: I find that code in https://github.com/elastic/elasticsearch/blob/674f7dc44c483ed55330144600618a7d220ede05/core/src/main/java/org/elasticsearch/common/settings/Settings.java#L161 and https://github.com/elastic/elasticsearch/blob/674f7dc44c483ed55330144600618a7d220ede05/core/src/main/java/org/elasticsearch/common/settings/Settings.java#L167 cause that bug, maybe it needs to be revised like this:
```
map.put(prefix + key, newMap); //L161
map.put(prefix + key, innerMap);//L167
```
then the bug is solved.

**Comment by williamrandolph** [comment]
@kkewwei Thank you for the detailed analysis and instructions to reproduce. I believe you've identified a definite bug, and your PR #44047 has the correct solution. I'll post some further discussion there.

**Comment by williamrandolph** [comment]
We've merged the fix to `master` and backported to the 7.3 and 7.x branches.

Thanks again to @kkewwei for the contribution.
